sdk/core/core-client/src/utils.ts

```diff
@@ -82,30 +146,34 @@ export function flattenResponse(
           arrayResponse[key] = parsedHeaders[key];
         }
       }
-      return arrayResponse;
+      return !fullResponse.parsedBody && isNullable && arrayResponse === emptyArray
```

Comment:
do we need the check of `arrayResponse === emptyArray`? Won't it always be true if `fullResponse.parsedBody` is falsy?

Reply:
we could have stuff in the headers :)

Reply:
Sure, but `===` in this case compares object identity. If you set keys on `emptyArray`, it will still return true here, yeah?

Reply:
You're right, I simplified it by dropping `!fullResponse.parsedBody` and used `getOwnPropertyNames` to check if the number of properties is 0 instead.